pkg/semver/semver.go

```diff
@@ -0,0 +1,517 @@
+// Copyright (C) 2014-2019, Matt Butcher and Matt Farina
+
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package semver
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
```

Comment:
What are modifications compared to the original code?

Reply:
Only the difference from the original is `sort.go`, for sorting by newer.

The main reason why I borrowed the code instead of import is to avoid being dependent. It has tiny code so I thought copying is enough.

Reply:
I think if the code was not modified so instead of copying it into our source base, importing it via go-mod is better (as other libs).
Updating to the new version of it becomes easier to do.

Btw, `sort.go` can be removed by using https://golang.org/pkg/sort/#Slice to sort the slice.

Reply:
Thanks. Okay, I'd be happy to import it and use sort.Slice :-)